### PR TITLE
docs(undici): Update option links to point to right link

### PR DESCRIPTION
**Pull request by AbhiPrasad**
## Which problem is this PR solving?

Current options table points to https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-undici

## Short description of the changes

Change to https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/instrumentation-undici and fix how line numbers are referenced

### CODE REVIEW

File: plugins/node/instrumentation-undici/README.md [hunk 1]
```diff
@@ -25,7 +25,9 @@ OpenTelemetry Undici/fetch Instrumentation allows the user to automatically coll
 To load a specific instrumentation (Undici in this case), specify it in the Node Tracer's configuration.
 
 ```js
-const { UndiciInstrumentation } = require('@opentelemetry/instrumentation-undici');
+const {
+  UndiciInstrumentation,
```
Comment (AbhiPrasad) [inline plugins/node/instrumentation-undici/README.md:29]: my prettier formatter did these changes, lmk if you want them to be reverted.

**Review by JacksonWeber — APPROVED** [review]

File: plugins/node/instrumentation-undici/README.md [hunk 2]
```diff
@@ -41,20 +43,19 @@ provider.register();
...
+| Options                                                                                                                                                | Type                    | Description                                                                                                                                                                                                                                                  |
+| ------------------------------------------------------------------------------------------------------------------------------------------------------ | ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
```
Comment (trentm) [inline plugins/node/instrumentation-undici/README.md:53]: You have a formatter that does this for you, do you?

nit: I'm not sure this helps readablility for those reading README.md in a text editor. ;)
However, I know why it is being done.

**Review by trentm — APPROVED** [review]

File: plugins/node/instrumentation-undici/README.md (hunk 1 above)
Comment (trentm) [reply plugins/node/instrumentation-undici/README.md:29]: All good.

File: plugins/node/instrumentation-undici/README.md (hunk 2 above)
Comment (AbhiPrasad) [reply plugins/node/instrumentation-undici/README.md:53]: like the code changes above, this was also formatted via a markdown extension for vscode that I have enabled by default for all JS related projects.

here's what this looks like in the plain diff

image

yeah unfortunately markdown formatting here is super subjective, I know other formatters take a different approach.

**Review by david-luna — APPROVED** [review]

**Pull request merged**